lib/src/main/java/io/ably/lib/types/Param.java

```diff
@@ -25,4 +25,23 @@ public static boolean containsKey(Param[] params, String key) {
 				return true;
 		return false;
 	}
+
+	@Override
+	public boolean equals(Object o) {
+		if (this == o) return true;
+		if (o == null || getClass() != o.getClass()) return false;
+
+		Param param = (Param) o;
+
+		if (key != null ? !key.equals(param.key) : param.key != null) return false;
+		return value != null ? value.equals(param.value) : param.value == null;
+
+	}
+
+	@Override
+	public int hashCode() {
+		int result = key != null ? key.hashCode() : 0;
+		result = 31 * result + (value != null ? value.hashCode() : 0);
+		return result;
+	}
```

Comment:
I am using them within HashSet, that requires all values to override both methods to ensure uniqueness.

https://github.com/ably/ably-java/pull/102/files#diff-db9d112187c909d918e250dd2544de78R482